server-common/src/main/java/org/apache/kafka/server/group/share/PersisterStateManager.java

```diff
@@ -550,6 +642,13 @@ private Node randomNode() {
      */
     @Override
     public Collection<RequestAndCompletionHandler> generateRequests() {
+      // we want to use the nodeRPC map and the queue to generate requests
+      // and maybe coalesce write requests for single node
+      if (generateCallback != null) {
```

Comment:
The comment above this line really doesn't do it justice :) Can we have a better comment please?